diff -Naur Zend.orig/Db/Table/Abstract.php Zend/Db/Table/Abstract.php
--- Zend.orig/Db/Table/Abstract.php	Thu Jun 19 11:21:02 2008
+++ Zend/Db/Table/Abstract.php	Sun Nov 30 13:12:32 2008
@@ -921,7 +921,12 @@
                                 $this->_db->quoteIdentifier($col, true) . ' = ?',
                                 $primaryKey[$refCol], $type);
                         }
-                        $rowsAffected += $this->delete($where);
+                        
+                        $toDelete = $this->fetchAll($where);
+                        foreach($toDelete as $row) {
+                            $rowsAffected += $row->delete();
+                        }
+                        
                         break;
                     default:
                         // no action
